File: QNote/QNote/vc/HomeViewController.m
//
//  HomeViewController.m
//  QNote
//
//  Created by 林宁宁 on 16/6/2.
//  Copyright © 2016年 林宁宁. All rights reserved.
//

#import "HomeViewController.h"
#import "HomeTableViewCell.h"

@interface HomeViewController ()<UITableViewDelegate,UITableViewDataSource>

@property (retain, nonatomic) NSMutableArray <NSDictionary *> * dataList;

@property (weak, nonatomic) IBOutlet UITableView *tableV;

@end

/**
 *  项目地址
 *  https://github.com/MaiyaT/AongQ
 */

@implementation HomeViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.dataList = [[NSMutableArray alloc] init];
        
    [self.dataList addObject:@{@"title":@"Runtime学习",@"subtitle":@"2016-6-23",@"vc":@"RunTimeViewController"}];
    
    [self.dataList addObject:@{@"title":@"HTML本地",@"subtitle":@"2016-6-23",@"vc":@"HTMLWebViewController"}];
    
    [self.dataList addObject:@{@"title":@"UIBezierPath",@"subtitle":@"2016-6-9",@"vc":@"BezierViewController"}];
    
    [self.dataList addObject:@{@"title":@"CAShapeLayer-Progress",@"subtitle":@"2016-6-7",@"vc":@"ProgressViewController"}];
    
    [self.dataList addObject:@{@"title":@"AFNetworking—学习",@"subtitle":@"2016-6-4__框架学习",@"vc":@"AFNetworkingNoteVC"}];
    
    [self.dataList addObject:@{@"title":@"AFNetworking—demo",@"subtitle":@"2016-6-4__AFNDemo",@"vc":@"AFNetworkingViewController"}];
    
    
    [self.dataList addObject:@{@"title":@"ValueForKeyPath",@"subtitle":@"2016-6-4__ValueForKeyPath操作",@"vc":@"ValueForKeyPathViewController"}];
    
    [self.dataList addObject:@{@"title":@"函数",@"subtitle":@"2016-6-3__数学函数",@"vc":@"MathFuntionViewController"}];
    
    
    [self.dataList addObject:@{@"title":@"PropertyKeyWords",@"subtitle":@"2016-6-2__属性的关键字",@"vc":@"PropertyKeyWordVController"}];
    
    
    [self.dataList addObject:@{@"title":@"NSSecureCoding",@"subtitle":@"2016-6-1__归档-编码解码",@"vc":@"SecureCodingViewController",@"sugueid":@"pushSecure"}];
    
    
    
    
    self.tableV.tableFooterView = [UIView new];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

-(CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
{
    return 60;
}

-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return self.dataList.count;
}

-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    HomeTableViewCell * cell = [tableView dequeueReusableCellWithIdentifier:@"noteCellIdentify"];
    
    NSDictionary * dataDic = self.dataList[indexPath.row];
    
    NSString * title = dataDic[@"title"];
    NSString * subtitle = dataDic[@"subtitle"];
    
    cell.cellTitle.text = title;
    cell.cellSubTitle.text = subtitle?subtitle:@"";
    
    return cell;
}

-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
    
    NSDictionary * dataDic = self.dataList[indexPath.row];
    
    NSString * title = dataDic[@"title"];
    NSString * vcClass   = dataDic[@"vc"];
    NSString * sugueid = dataDic[@"sugueid"];
    
    if(sugueid)
    {
        dispatch_async(dispatch_get_main_queue(), ^{
            [self performSegueWithIdentifier:sugueid sender:self];
        });
    }
//    else if ([[self storyboard] instantiateViewControllerWithIdentifier:vcClass])
//    {
//        [self.navigationController pushViewController:[[self storyboard] instantiateViewControllerWithIdentifier:vcClass] animated:YES];
//    }
    else
    {
        UIViewController * vc = [[NSClassFromString(vcClass) alloc] init];
        vc.title = title;
        [self.navigationController pushViewController:vc animated:YES];
    }
}

//判断 segue跳不跳
-(BOOL)shouldPerformSegueWithIdentifier:(NSString *)identifier sender:(id)sender
{
    return YES;
}

-(void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: YHNote/YHNote/VC/HomeViewController.swift
//
//  HomeViewController.swift
//  YHNote
//
//  Created by 林宁宁 on 16/7/11.
//  Copyright © 2016年 林宁宁. All rights reserved.
//

import UIKit

class HomeViewController: BaseViewController,UITableViewDelegate,UITableViewDataSource {
    
    var tableview : UITableView!
    var list: NSArray!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    
        self.title = "Swift"
        self.view.backgroundColor = UIColor.lightGrayColor()
        
        
        self.tableview = UITableView(frame: self.view.bounds, style: UITableViewStyle.Plain)
        tableview.delegate = self
        tableview.dataSource = self
        tableview.registerClass(UITableViewCell.self, forCellReuseIdentifier: "homeTableViewCellIdentify")
        self.view.addSubview(tableview)
        

        list = [
            ["title":"基础知识","vc":"SwiftBaseNoteViewController"],
            ["title":"目录结构-沙盒机制","vc":"DirectoryStructureViewController"],
            ["title":"数据持久化","vc":"DataPersistenceViewController"],
            ["title":"GCD","vc":"GCDExampleViewController"],
            ["title":"Encryption加密","vc":"SectureViewController"],
            ["title":"Popping动画","vc":"PoppingAnimationViewCtrl"]
            ["title":"RunLoop学习","vc":"RunLoopViewController"]
        ].copy() as! NSArray
        
    }

    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return list.count
    }
    
    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        return 1
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        
        let cell = tableview.dequeueReusableCellWithIdentifier("homeTableViewCellIdentify", forIndexPath: indexPath) as UITableViewCell
     
        let dic = list[indexPath.row] as? NSDictionary
        
        cell.textLabel?.text = dic!["title"] as? String
        
        return cell
    }
    
    
    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        
        let dic = list[indexPath.row] as? NSDictionary
        let vcStr = dic!["vc"] as? String
        
        let appName = NSBundle.mainBundle().infoDictionary!["CFBundleExecutable"] as! String
        
        let vcClass : AnyClass = NSClassFromString(appName + "." + vcStr!)!
        
        let vc = (vcClass as! UIViewController.Type).init()
        
        vc.title = dic!["title"] as? String
        
        self.navigationController?.pushViewController(vc, animated: true)
        
        
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
